### PR TITLE
Update the GraphQL Tool Guide with SDL Schema Generation

### PR DESCRIPTION
**Pull request by DimuthuMadushan**
Fixes: [#3818](https://github.com/ballerina-platform/ballerina-standard-library/issues/3818)

## Checklist

- [ ] **Page addition**
  - [ ] Add `permalink` to pages.

- [ ] **Page removal**
  - [ ] Remove entry from corresponding left nav YAML file.
  - [ ] Add `redirect_from` on the alternative page.
  - [ ] If no alternative page, add redirection on the `redirections.js ` file.

- [ ] **Page rename**
  - [ ] Add front-matter `redirect_from`.
  - [ ] Add front-matter `redirect_to:` (if applicable).

- [ ] **Page restrcuture**
  - [ ] Add `permalink` to pages.
  - [ ] Add front-matter `redirect_from`.
  - [ ] Add front-matter `redirect_to:` (if applicable).

### CODE REVIEW

File: swan-lake/learn-the-platform/ballerina-tooling/graphql-tool.md [hunk 1]
```diff
@@ -1,38 +1,53 @@
...
-description: Check out how the Ballerina GraphQL client tool makes it easy for you to generate a client in Ballerina using a given GraphQL schema (SDL) and GraphQL queries.
+title: GraphQL tool
+description: Check out how the Ballerina GraphQL tool makes it easy for you to generate a client in Ballerina using a given GraphQL config file and a GraphQL schema using a given Ballerina GraphQL service.
 keywords: ballerina, programming language, graphql, sdl, schema definition language
-permalink: /learn/graphql-client-tool/
-active: graphql-client-tool
-intro: The Ballerina GraphQL client tool makes it easy for you to generate a client in Ballerina using a given GraphQL schema (SDL) and GraphQL queries. 
+permalink: /learn/graphql-tool/
+active: graphql-tool
+intro: The Ballerina GraphQL tool will make it easy for you to start the development of a GraphQL client. It generates the client skeletons in Ballerina for a given GraphQL schema and a GraphQL document. In addition, the GraphQL tool supports generating the GraphQL schema specified by the Schema Definition Language for a given Ballerina GraphQL service.
```
Comment (ThisaruGuruge) [inline swan-lake/learn-the-platform/ballerina-tooling/graphql-tool.md:8]:
```suggestion
intro: The Ballerina GraphQL tool will make it easy for you to start the development of a GraphQL client. It generates the client skeletons in Ballerina for a given GraphQL schema and a GraphQL document. In addition, the GraphQL tool supports generating the GraphQL schema for a given Ballerina GraphQL service and write it to a file in GraphQL schema definition language(SDL).
```

File: swan-lake/learn-the-platform/ballerina-tooling/graphql-tool.md [hunk 2]
```diff
@@ -1,38 +1,53 @@
...
 
+The Ballerina GraphQL tooling support provides the following capabilities.
+
+1. Generating a Ballerina client from a given GraphQL config file configured with a GraphQL schema specified by Schema Definition Language and a GraphQL document.
+
+2. Generating a Ballerina client from a given GraphQL config file configured with a GraphQL schema specified by Schema Definition Language and multiple GraphQL documents.
+
+3. Generating multiple Ballerina modules from a given GraphQL config file configured with multiple GraphQL projects.
+
+4. Generate the GraphQL schema specified by the Schema Definition Language for a given Ballerina GraphQL service(s).
```
Comment (ThisaruGuruge) [inline swan-lake/learn-the-platform/ballerina-tooling/graphql-tool.md:24]: This doesn't make sense.

```suggestion
4. Generate the GraphQL schema for a given Ballerina GraphQL service(s) and write the schema(s) to a file/files using GraphQL schema definition language.
```

File: swan-lake/learn-the-platform/ballerina-tooling/graphql-tool.md [hunk 3]
```diff
@@ -244,4 +259,44 @@ This generates a separate Ballerina module (with the project name configured) fo
...
\ No newline at end of file
+You can use the generated modules to work with multiple GraphQL APIs. Use the client that is generated within each module to execute the queries/mutations on the relevant GraphQL API. Also, use the clients that are generated **within** the same Ballerina package by importing the generated modules. For more information, see [Import a module from the same package](/learn/organize-ballerina-code/#import-a-module-from-the-same-package).
+
+## Schema Generation
+
+The GraphQL tool supports generating the GraphQL schema specified by the Schema Definition Language for a given Ballerina GraphQL service. You will be able to export the generated schema to the given location. The tool can be used in the following ways to generate the schema.
```
Comment (ThisaruGuruge) [inline swan-lake/learn-the-platform/ballerina-tooling/graphql-tool.md:266]: The schema is not specified by the Schema Definition Language. In Ballerina, schema is defined using the Ballerina service. Furthermore, SDL is a generic term, not specific to GraphQL. Shall we reword this accordingly?

```suggestion
The GraphQL tool supports generating the GraphQL schema specified by the Schema Definition Language for a given Ballerina GraphQL service. You will be able to export the generated schema to the given location. The tool can be used in the following ways to generate the schema.
```

**Review by sm1990 — APPROVED** [review]

**Review by ThisaruGuruge — APPROVED** [review]

File: swan-lake/learn-the-platform/ballerina-tooling/graphql-tool.md [hunk 4]
```diff
@@ -87,7 +102,7 @@ You can validate and execute the GraphQL queries using the >GraphQL Foundation V
 
 >**Note:** You are required to give a query name for each query. This name is used as the corresponding remote operation name in the generated client.
```
Comment (praneesha) [inline swan-lake/learn-the-platform/ballerina-tooling/graphql-tool.md:103]:
```suggestion
>**Note:** You are required to give a name for each query. This name is used as the corresponding remote operation name in the generated client. 
```

File: swan-lake/learn-the-platform/ballerina-tooling/graphql-tool.md [hunk 5]
```diff
@@ -244,4 +259,44 @@ This generates a separate Ballerina module (with the project name configured) fo
 2. `utils.bal` - contains all the utility methods related to the client stub
 3. `types.bal`- contains all the Ballerina data types extracted from the GraphQL schema (SDL)
 
-You can use the generated modules to work with multiple GraphQL APIs. Use the client that is generated within each module to execute the queries/mutations on the relevant GraphQL API. Also, use the clients that are generated **within** the same Ballerina package by importing the generated modules. For more information, see [Import a module from the same package](/learn/organize-ballerina-code/#import-a-module-from-the-same-package).
\ No newline at end of file
+You can use the generated modules to work with multiple GraphQL APIs. Use the client that is generated within each module to execute the queries/mutations on the relevant GraphQL API. Also, use the clients that are generated **within** the same Ballerina package by importing the generated modules. For more information, see [Import a module from the same package](/learn/organize-ballerina-code/#import-a-module-from-the-same-package).
+
+## Schema Generation
```
Comment (praneesha) [inline swan-lake/learn-the-platform/ballerina-tooling/graphql-tool.md:264]:
```suggestion
## Schema generation
```

File: swan-lake/learn-the-platform/ballerina-tooling/graphql-tool.md [hunk 6]
```diff
@@ -244,4 +259,44 @@ This generates a separate Ballerina module (with the project name configured) fo
...
+
+```
+$ bal graphql -i service.bal -o ./schema
+```
+
+The `input` parameter specifies the path of the ballerina service file (e.g., `service.bal`) and is mandatory.
+
+>**Info:** For more information on the command, see [Ballerina to GraphQL](/learn/cli-documentation/graphql/#ballerina-to-graphql).
+
+### Generate a schema for a specific service.
```
Comment (praneesha) [inline swan-lake/learn-the-platform/ballerina-tooling/graphql-tool.md:286]:
```suggestion
### Generate a schema for a specific service
```

File: downloads/swan-lake-release-notes/swan-lake-2201.1.0/RELEASE_NOTE.md [hunk 7]
```diff
@@ -636,7 +636,7 @@ Introduced the AsyncAPI tool, which will make it easy for you to start the devel
 
 #### GraphQL client tool
 
-Introduced the GraphQL client tool, which will make it easy for you to generate a client in Ballerina given the GraphQL schema (SDL) and GraphQL queries. Ballerina Swan Lake supports the GraphQL specification [October 2021 edition](https://spec.graphql.org/October2021/). For more information, see [Ballerina GraphQL support](/learn/graphql-client-tool/) and [GraphQL CLI documentation](/learn/cli-documentation/graphql/#graphql-to-ballerina).
+Introduced the GraphQL client tool, which will make it easy for you to generate a client in Ballerina given the GraphQL schema (SDL) and GraphQL queries. Ballerina Swan Lake supports the GraphQL specification [October 2021 edition](https://spec.graphql.org/October2021/). For more information, see [Ballerina GraphQL support](/learn/graphql-tool/) and [GraphQL CLI documentation](/learn/cli-documentation/graphql/#graphql-to-ballerina).
```
[the remark this answers is not in the document]
Comment (DimuthuMadushan) [reply downloads/swan-lake-release-notes/swan-lake-2201.1.0/RELEASE_NOTE.md:639]: This is a release note that was added previously. I changed only the link that directs to the GraphQL tool. Is it ok to change this release note?

File: swan-lake/learn-the-platform/ballerina-tooling/graphql-tool.md [hunk 8]
```diff
@@ -1,38 +1,53 @@
 ---
 layout: ballerina-graphql-support-left-nav-pages-swanlake
-title: GraphQL client tool
-description: Check out how the Ballerina GraphQL client tool makes it easy for you to generate a client in Ballerina using a given GraphQL schema (SDL) and GraphQL queries.
+title: GraphQL tool
+description: Check out how the Ballerina GraphQL tool makes it easy for you to generate a client in Ballerina using a given GraphQL config file and a GraphQL schema using a given Ballerina GraphQL service.
```
[the remark this answers is not in the document]
Comment (DimuthuMadushan) [reply swan-lake/learn-the-platform/ballerina-tooling/graphql-tool.md:4]: This changes the idea. There are two tasks that can be done using the GraphQL tool,
1. generate a client in Ballerina using a given GraphQL config file.
2. generate a GraphQL schema using a given Ballerina GraphQL service.
```suggestion
description: Check out how the Ballerina GraphQL tool makes it easy for you to generate a client in Ballerina using a given GraphQL config file and generate a GraphQL schema using a given Ballerina GraphQL service.
```

File: downloads/swan-lake-release-notes/swan-lake-2201.1.0/RELEASE_NOTE.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (praneesha) [reply downloads/swan-lake-release-notes/swan-lake-2201.1.0/RELEASE_NOTE.md:639]: It is fine to change the release note as well.

File: swan-lake/learn-the-platform/ballerina-tooling/graphql-tool.md (hunk 8 above)
[the remark this answers is not in the document]
Comment (praneesha) [reply swan-lake/learn-the-platform/ballerina-tooling/graphql-tool.md:4]: +1 for this.

**Review by praneesha — APPROVED** [review]

File: downloads/swan-lake-release-notes/swan-lake-2201.1.0/RELEASE_NOTE.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (ThisaruGuruge) [reply downloads/swan-lake-release-notes/swan-lake-2201.1.0/RELEASE_NOTE.md]: I don't think changing the text of a previous release note is a good practice. If there are issues in the text, they should've been identified and fixed before the release. Changing the release note months after a release can have negative impacts. (Not in this case, of course, but it is not a good practice at all, IMO).

**Pull request merged**